### ISSUE TITLE
SqlServer: dropIfExists fails if the Slick schema has new foreign keys.

### ISSUE THREAD

**Issue by akozumpl** [slick/slick#2118]
**DESCRIPTION:**

In SqlServer, running `table.dropIfExists` fails with an exception if the table exists in the DB but does not contain one or more foreign keys which the current schema defines.

I frequently encounter this problem when switching between development branches with slightly different table schema and running `dropIfExists` for setup.

**STEPS TO REPRODUCE:**

Create a table A and a table B. 

Write a Slick table definition for B but add a new FK.

Run `b.dropIfExists()`.

**EXPECTED RESULT:**

`dropIfExists` shouldn't throw and it should ideally drop the named table despite imperfect schema match.

**ACTUAL RESULT:**

Will fail with an exception:

```
com.microsoft.sqlserver.jdbc.SQLServerException: 'FK_S2_T' is not a constraint.
```

**SETUP:**

SqlServer 2016

**Comment by akozumpl** [comment]
Possibly related: #2063.

**Comment by nafg** [comment]
I guess this was fixed in #2119
